check-ntpoffset/lib/check-ntpoffset.go

```diff
@@ -51,14 +51,20 @@ func run(args []string) *checkers.Checker {
 
 	var chkSt checkers.Status
 	var msg string
+	stratum, err := getNTPStratum()
+	if err != nil {
+		msg = fmt.Sprintf("ntp stratum not found offset is %f(actual), %f(warning threshold), %f(critical threshold)", math.Abs(offset), opts.Warn, opts.Crit)
+		return checkers.NewChecker(checkers.WARNING, msg)
+	}
+
 	if opts.Crit < math.Abs(offset) {
 		msg = fmt.Sprintf("ntp offset is over %f(actual) > %f(threshold)", math.Abs(offset), opts.Crit)
 		chkSt = checkers.CRITICAL
 	} else if opts.Warn < math.Abs(offset) {
 		msg = fmt.Sprintf("ntp offset is over %f(actual) > %f(threshold)", math.Abs(offset), opts.Warn)
 		chkSt = checkers.WARNING
 	} else {
-		msg = fmt.Sprintf("ntp offset is %f(actual) < %f(warning threshold), %f(critial threshold)", math.Abs(offset), opts.Warn, opts.Crit)
+		msg = fmt.Sprintf("ntp stratum=%s, offset is %f(actual) < %f(warning threshold), %f(critial threshold)", stratum, math.Abs(offset), opts.Warn, opts.Crit)
```

Comment:
Should the stratum name be added to the other messages above?

Reply:
I removed the log message about stratum.